waka_waka/Scenes/maze.gd

```diff
@@ -0,0 +1,256 @@
+extends TileMapLayer
+
+@export var maze_width: int = 90
+@export var maze_height: int = 90
+
+const DIRECTIONS: Array = [
+	Vector2i(0, -1), # North
+	Vector2i(1, 0),  # East
+	Vector2i(0, 1),  # South
+	Vector2i(-1, 0)  # West
+]
+
+# Lookup table: maps hashed PackedByteArray (NESW) to (atlas_id, # of clockwise rotations)
+var TILE_LOOKUP = {
+	hash(PackedByteArray([0, 0, 0, 0])): { "id": Vector2i(0, 0), "rotation": Vector2i(0, -1) },  # Isolated wall (Island)
+	hash(PackedByteArray([0, 1, 0, 1])): { "id": Vector2i(1, 0), "rotation": Vector2i(0, -1) },  # Horizontal wall (East-West)
+	hash(PackedByteArray([1, 0, 1, 0])): { "id": Vector2i(1, 0), "rotation": Vector2i(1, 0) },  # Vertical wall (North-South)
+	hash(PackedByteArray([1, 1, 0, 0])): { "id": Vector2i(2, 0), "rotation": Vector2i(0, -1) },  # Outer corner (North and East)
+	hash(PackedByteArray([0, 1, 1, 0])): { "id": Vector2i(2, 0), "rotation": Vector2i(1, 0) },  # Outer corner (East and South)
+	hash(PackedByteArray([0, 0, 1, 1])): { "id": Vector2i(2, 0), "rotation": Vector2i(0, 1) },  # Outer corner (South and West)
+	hash(PackedByteArray([1, 0, 0, 1])): { "id": Vector2i(2, 0), "rotation": Vector2i(-1, 0) },  # Outer corner (West and North)
+	hash(PackedByteArray([1, 1, 1, 1])): { "id": Vector2i(3, 0), "rotation": Vector2i(0, -1) },  # Cross intersection
+	hash(PackedByteArray([1, 0, 0, 0])): { "id": Vector2i(4, 0), "rotation": Vector2i(0, -1) },  # Endcap (North)
+	hash(PackedByteArray([0, 1, 0, 0])): { "id": Vector2i(4, 0), "rotation": Vector2i(1, 0) },  # Endcap (East)
+	hash(PackedByteArray([0, 0, 1, 0])): { "id": Vector2i(4, 0), "rotation": Vector2i(0, 1) },  # Endcap (South)
+	hash(PackedByteArray([0, 0, 0, 1])): { "id": Vector2i(4, 0), "rotation": Vector2i(-1, 0) },  # Endcap (West)
+	hash(PackedByteArray([1, 1, 0, 1])): { "id": Vector2i(5, 0), "rotation": Vector2i(0, -1) },  # T-junction (no South)
+	hash(PackedByteArray([1, 1, 1, 0])): { "id": Vector2i(5, 0), "rotation": Vector2i(1, 0) },  # T-junction (no East)
+	hash(PackedByteArray([0, 1, 1, 1])): { "id": Vector2i(5, 0), "rotation": Vector2i(0, 1) },  # T-junction (no North)
+	hash(PackedByteArray([1, 0, 1, 1])): { "id": Vector2i(5, 0), "rotation": Vector2i(-1, 0) }   # T-junction (no West)
+}
+
+var tile_transforms: Dictionary = {
+	Vector2i(0, -1): 0, # No rotation, default orientation
+	Vector2i(1, 0): TileSetAtlasSource.TRANSFORM_TRANSPOSE | TileSetAtlasSource.TRANSFORM_FLIP_H, # 90 degrees clockwise
+	Vector2i(0, 1): TileSetAtlasSource.TRANSFORM_FLIP_H | TileSetAtlasSource.TRANSFORM_FLIP_V, # 180 degrees clockwise
+	Vector2i(-1, 0): TileSetAtlasSource.TRANSFORM_FLIP_V | TileSetAtlasSource.TRANSFORM_TRANSPOSE # 270 degrees clockwise
+}
+
+var piece_size: int = 3
+
+var tiles: Dictionary = {
```

Comment:
Does this define the metashape? You stack these however you want and then fill them with something chosen from the hash(PackedByteArray from above)

Reply:
Basically right now it's two steps =>
1. Create "coarse_grid" 1/3 the side of final maze, fill it with random letters from the dict.
2. Create the "final_grid" by expanding each letter to the tile, and randomly rotating it (0, 90, 180, 270) degrees.